contracts/MemeFactory.sol

```diff
@@ -380,6 +386,10 @@ abstract contract MemeFactory {
         // Check the unleash timestamp
         require(block.timestamp >= memeSummon.summonTime + UNLEASH_DELAY, "Cannot unleash yet");
 
+        // Wrap native token to its ERC-20 version
```

Comment:
This has to be here. With the collect fee logic we otherwise have an error. Wrapping all here makes it trivial